### ISSUE TITLE
Make KWoC less stressful for mentors

### ISSUE THREAD

**Issue by OrkoHunter** [kossiitkgp/kwoc-2017#143]
- If a mentor registers more than 2 projects, contact them and say "You might get stressed because of some low-quality patches, as we have seen in the past. Hence we request you to kindly submit at max 2 projects".

 - Instead of suggesting, "Choose a project, contact mentor and contribute", we should say "Choose a project, read our student's manual(link it). We strongly expect you to proceed after reading the manual and following the instructions written in it. Read the project's contributors' guideline. Find some issues and contribute."

 - When a mentor submits a project, ask them to set up "contributors' guideline" for their project and reference it from their README. If they don't have it, help them set up from the examples [[1]](https://help.github.com/articles/setting-guidelines-for-repository-contributors/) [[2]](https://gist.github.com/PurpleBooth/b24679402957c63ec426).

- Do not make mentor's email id public, a communication channel/forum should be the only way to contact them (like GSoC orgs). Personal emails are supposed to be ignored. If the mentor doesn't feel like setting up a forum and prefers personal emails, they can put their email id in the communication channel link. No big deal.

 - Request mentors to add a code of conduct to their repositories, which is mandatory to be read by every contributor of their repository. Even after a warning, if a student misbehaves, they can surely be banned from the repository. (This is perfectly fine and happens in organizations with very big impacts e.g. sympy)

 - Communicate to students more. Put up learning materials like https://www.learnitgirl.com/ -> Learning materials. Send it to them. Also, send them basic FAQs at the beginning, some of them are
Q: "How do I get started with a project?"
A: (to be elaborated) Set up the repository (See github guidelines) and use the project. Read their contributor's guidelines or a-link-to-template-contributor's-guidelines. (How about [opensource.guide](https://opensource.guide) articles ? ) See the open issues and try to fix them. You can comment on the issue if you think you have searched the internet enough for solutions and read the related materials/documentation completely) 

(Open for discussions)

**Comment by pranitbauva1997** [comment]
All of this is great! But the main underlying issue is that most students are not at all wiling to even read a document. There have been many times, I have pointed out to a document and got only 5% rate of reading which is pretty sad. Sorry if I sound ranting, but this is the **big issue** and right now I have very little idea about how to solve it.

**Comment by OrkoHunter** [comment]
I agree with you @pranitbauva1997. That is a big problem. Even I read the GSoC student's manual while writing one for KWoC last year. But it turned out that I already knew a bunch of the things written inside it because of reading things from here and there (READMEs, forums, mentor's previous emails in the forum, a hell lot of contributor's guidelines, emails from GSoC etc.)
Hence, instead of just putting up `Manuals` on the website, we need to put up good bits of the instructions all over the places.
The new generation of students seems to be heavily influenced by Quora (Pinging @djokester to get involved in the discussion). Maybe we can put up some FAQs there. Surprisingly new people tend to read Quora answers more and pdfs less.

**Comment by pranitbauva1997** [comment]
@OrkoHunter This is a great idea. We can have a standard reply crafted and for all KWoC related Quora question, we will stick that along with the other words by someone who is actually writing the answer. What do you think @djokester?

**Comment by kaustubhhiware** [comment]
To communicate this from next time on, a code of conduct post must be made on KOSS's blog as well, regardless of whether a Quora answer is put up or not.

**Comment by OrkoHunter** [comment]
Yes @kaustubhhiware, along with a blog, we can put a good referencable link for the Code of Conduct on the website itself.

**Comment by djokester** [comment]
What we can do is a few of us host Quota sessions every weekend a month or
two before KWoC. This enables us to get a wide range of questions. We can
choose wisely and answer those to the best of our abilities.
A second thing is that most of the students is not worth investing our
times upon. It's bad to say this as a mentor but I find a lot of students
to be a lot of talk and no work. They just never find a starting point.
They keep on pestering someone or the other but there is absolutely zero
output. I think this year's student evaluation will reflect that upon me as
a mentor.

**Comment by dhruvapte26** [comment]
I found @djokester's idea of a Quora session partially effective. Despite the idea being attractive, I noticed that not a lot of people ask questions in these sessions which are related(see this [answer](https://www.quora.com/How-do-you-feel-when-you-%E2%80%9Ctake-questions%E2%80%9D-on-Quora/answer/Samriddhi-Sinha?srid=CnjP) written by @djokester himself). Though there is a topic of KWoC on Quora, not a lot of people are reading answers given by some of the mentors

**Comment by djokester** [comment]
See the point here is completely different. Kindly do not bring up answers
that do not reflect what I want to.point out.
I am known on Quota for two reasons. IIT Kharagpur and GSoC.
So any session relayed to the two of these will be successful. I received
more than 12 questions and all of them related to KWoC when I hosted a
session on KWoC. I A2Aed them to @OrkoHunter.
I have no issues with Open Source and JEE related topics. I ranted in that
question because my followers don't let me expand my topics of expertise.

**Comment by dhruvapte26** [comment]
I apologize for bringing up such a thing. What I wanted to point out is that people might not ask specifics as related to the session. :smile: . As you mentioned about the questions, we can, um possibly use those answer links as a 'guide' instead of the original PDF ones. This might generate more interest among the contributors instead.

**Comment by OrkoHunter** [comment]
@the-ethan-hunt If you read the first comment on this issue mentioning Quora, it is not about the type of questions people ask, it is that new people read Quora with more trust and attention. Hence it is irrelevant to what people would ask because we already know most of the questions. Hence we ourselves will put our frequently asked questions and get it answered over there.

New people joining the thread: Please read the points mentioned above carefully and comment your criticism with brevity.

**Comment by icyflame** [comment]
I don't think you should cap mentor projects at 2. I submitted 3 programming projects (Node.js) and 1 non-programming project. (awesome-social-science) I recd a **LOT** of patches on the non-programming one. If asked to choose any two, I would probably not have chosen the non-programming one.

> I find a lot of students to be a lot of talk and no work. They just never find a starting point.

True also. Several people who mailed me the standard "please tell me how to contribute in detail" never did anything. It's not a bad thing, it's not annoying, it's not unexpected. But it reflects badly on the program. If I have the chance to register as a mentor again, I won't register any programming projects at all.

Everything else I wanted to say has already been covered by others here. @OrkoHunter thanks for spending time on this, I am sure this program will get better each year :+1:

**Comment by kshitij10496** [comment]
Talking with @ghostwriternr and @icyflame, we came up with the suggestion of having the student upload a 1-2 page Proposal about what they want to work on during KWoC.

**Advantages**:
- Project timeline for the student so that they can accommodate time for their exams.
- Structure to the mentor's schedule.
- Provides the mentor with a metric to keep track of the student's progress, thus, helping in evaluations.
- Motivates the student to patiently read about the project instead of repeatedly asking the mentor to help him out with trivial questions.

**Comment by kaustubhhiware** [comment]
Regarding the mentor's evaluations, it should be possible to rate mentee's work on a scale, rather than a boolean Pass or Fail.
2 mentees worked with me before mid-term evaluation, one made only one PR (merged) on the very first day, and the second made 3-4 PRs (all merged). It would be unfair to the second student to be rated equally as the first one.

Up to you if you want to accommodate this.

**Comment by dhruvapte26** [comment]
- I discussed with @kshitij10496 about his proposal suggestion. We discussed the issue that maybe instead of asking petty questions over 'how to contribute', we might get questions related to 'how to draft a proposal'. In that case, he suggested a sample proposal. This seems a really cool option. :smile:
- Secondly, I would like to suggest *3-4 evaluation procedures*. Some of my mentees worked really well before mid-term and then randomly disappeared without notice. If we have 3 evaluations(let's say over a period of 8-10 days) we can ensure more effective participation. The scoring system can be as @kaustubhhiware  proposed.
- A significant issue came up this year when several newcomers tried to 'take-over' the project. Some despite being warned did this repeatedly. We can here have something like "three strikes and you are Out!" where the mentee is warned twice and on the third time, he/she is not allowed to participate in KWoC, let alone being banned from the project. This serves as stricter than banning and lenient than reporting and can serve as a procedure for other misconduct that might/may happen during KWoC.


Criticism and reviews welcomed for these suggestions :smile:
